### PR TITLE
🌱 Bump kubebuilder-release-tools to v0.3.0, start running it on cherry-pick PRs

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Büringer buringerst@vmware.com

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: .github/workflows/verify.yml [hunk 1]
```diff
@@ -14,9 +14,6 @@ jobs:
     steps:
     - name: Verifier action
       id: verifier
-      uses: kubernetes-sigs/kubebuilder-release-tools@v0.2.0
-      # Don't run this step on PRs created by the cherry-pick bot,
-      # as the PR description is too short, but otherwise the PRs are fine.
```
Comment (sbueringer) [inline .github/workflows/verify.yml:19]: The descriptiveness check was dropped: https://github.com/kubernetes-sigs/kubebuilder-release-tools/pull/49

**Review by fabriziopandini — COMMENTED** [review]
/lgtm
/approve

**Comment by vincepri** [comment]
whoops :D commented on a old diff, ignore!

**Comment by sbueringer** [comment]
I only see the lgtm approve, all good :)

**Pull request merged**